lib/acts-as-taggable-array-on/taggable.rb

```diff
@@ -20,17 +20,33 @@ def acts_as_taggable_array_on(tag_name, *)
 
         self.class.class_eval do
           define_method :"all_#{tag_name}" do |options = {}, &block|
-            subquery_scope = unscoped.select("unnest(#{table_name}.#{tag_name}) as tag").distinct
-            subquery_scope = subquery_scope.instance_eval(&block) if block
-            # Remove the STI inheritance type from the outer query since it is in the subquery
-            unscope(where: :type).from(subquery_scope).pluck(:tag)
+            # Handles the unique case of prepending method with "where("tag like ?", "aws%")"
+            missing_like_tag_prepend = current_scope&.where_clause&.send(:predicates)&.none? { |pred| pred.to_s.include?("tag like") }
```

Comment:
**issue:** we're using a protected `:predicates` here which is internal to AR and could change at any moment. This rubs me in a wrong way... as a code that potentially could cause issues.

I've been brainstorming for a solution without private/protected methods and only managed to come up with a following:
```suggestion
            missing_like_tag_prepend = !current_scope&.where_clause.any? { |t| t.is_a?(String) && t.include?("tag like") }

```